general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -920,6 +933,55 @@ def __repr__(self) -> str:
         )
 
 
+def read_ibm_credentials(ibmq_name: str | None) -> dict[str, str]:
+    """Function to try to read IBM credentials from .qiskit/qiskit-ibm.json.
+
+    Args:
+        ibmq_name: The name under which the IBM account credentials are locally stored.
+
+    Raises:
+        FileNotFoundError: If the configuration file is not found.
+        KeyError: If the provided `ibmq_name` does not have credentials stored in the
+            config file or `token` and/or `channel` keys are missing for the credentials
+            of the account under `ibmq_name`.
+        ValueError: If no `ibmq_name` is provided and multiple accounts are found with
+            none marked as default.
+
+    Returns:
+        Dictionary containing the ibm token, channel, and instance (if available).
+    """
+    config_dir = pathlib.Path.home().joinpath(".qiskit")
+    path = config_dir.joinpath("qiskit-ibm.json")
+    if path.is_file():
+        config = json.load(open(path))
+        if ibmq_name is None:
+            if len(config) == 1:
+                ibmq_name = list(config.keys())[0]
+            elif any(creds.get("is_default_account") for creds in config.values()):
+                ibmq_name = next(name for name in config if config[name].get("is_default_account"))
+            else:
+                raise ValueError(
+                    "Multiple accounts found but none are marked as default.",
+                    " Please provide the name of the account to retrieve.",
+                )
+        elif ibmq_name not in config:
+            raise KeyError(
+                f"No account credentials saved under the name {ibmq_name}"
+                f" in the config file found at {path}."
+            )
+
+        credentials = config.get(ibmq_name)
+        if any(key not in credentials for key in ["token", "channel"]):
+            raise KeyError(
+                "`token` and/or `channel` keys missing from credentials for the account",
+                f" under the name {ibmq_name} in the file {path}.",
```

Comment:
nits: wrap line at 100 and add quotes:
```suggestion
                "`token` and/or `channel` keys missing from credentials for the account under the "
                f"name '{ibmq_name}' in the file '{path}'.",
```